migrations/migrations/0019_calc_mediafile_is_public.py

```diff
@@ -0,0 +1,202 @@
+            self.mediafiles[id_].get("access_group_ids"),
+            parent_is_public,
+            parent_inherited_access_group_ids,
+        )
+        changed: bool = False
+        if calc_is_public != self.mediafiles[id_].get(
+            "is_public"
+        ) or calc_inherited_access_group_ids != self.mediafiles[id_].get(
+            "inherited_access_group_ids"
+        ):
+            changed = True
+            add = set(calc_inherited_access_group_ids) - set(
+                self.mediafiles[id_].get("inherited_access_group_ids", set())
+            )
+            if add:
+                self.__add_to_group(add, id_, True)
+            remove = set(
+                self.mediafiles[id_].get("inherited_access_group_ids", set())
+            ) - set(calc_inherited_access_group_ids)
+            if remove:
+                self.__add_to_group(remove, id_, False)
+            self.mediafiles[id_]["is_public"] = calc_is_public
+            self.mediafiles[id_][
+                "inherited_access_group_ids"
+            ] = calc_inherited_access_group_ids
+            update_event = UpdateEvent(
+                fqid,
+                {
+                    "is_public": calc_is_public,
+                    "inherited_access_group_ids": calc_inherited_access_group_ids,
+                },
+            )
+            self.events.append(update_event)
+
+        for child_id in self.mediafiles[id_].get("child_ids", []) or []:
+            if changed or child_id in self.mediafiles.keys():
+                self.check_recursive(child_id)
+
+    def __add_to_group(self, group_ids: Set[int], mediafile_id: int, add: bool) -> None:
+        for group_id in group_ids:
+            for group_id in self.groups:
+                value_list = set(
+                    self.groups[group_id]["add" if add else "remove"][
+                        "mediafile_inherited_access_group_ids"
+                    ]
+                )
+                value_list.add(mediafile_id)
+                self.groups[group_id]["add" if add else "remove"][
+                    "mediafile_inherited_access_group_ids"
+                ] = list(value_list)
+            else:
+                self.groups[group_id] = {
+                    "add": {
+                        "mediafile_inherited_access_group_ids": [mediafile_id]
+                        if add
+                        else []
+                    },
+                    "remove": {
+                        "mediafile_inherited_access_group_ids": []
+                        if add
+                        else [mediafile_id]
+                    },
+                }
+
+
+def calculate_inherited_groups_helper(
+    access_group_ids: Optional[List[int]],
+    parent_is_public: Optional[bool],
+    parent_inherited_access_group_ids: Optional[List[int]],
+) -> Tuple[bool, List[int]]:
+    """
+    TODO: If migration and backend are combined one day,
+    import this function from openslides_backend/models/helper.py
```

Comment:
Thought again about this and I think it's actually good to keep the migration separated from the actual code. So I would just remove this TODO.

Reply:
done